src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductServiceImpl(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product create(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product getById(Long id) {
+        return productRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException("Product by id " + id + " doesn't exist")
+        );
+    }
+
+    @Override
+    public void delete(Long id) {
+        productRepository.deleteById(id);
+    }
+
+    @Override
+    public Product update(Product product) {
+        Long id = product.getId();
+        Optional<Product> optionalProduct = productRepository.findById(id);
+        if (optionalProduct.isPresent()) {
+            return productRepository.save(product);
+        }

```

Comment:
could do with orElseThrow 
```suggestion
        return productRepository.findById(product.getId())
                .map(p -> productRepository.save(product))
                .orElseThrow(() -> new NoSuchElementException("Product by id " + product.getId() + " doesn't exist"));
```